### ISSUE TITLE
Internal server error with graphql queries

### ISSUE THREAD

**Issue by jwuclorox** [pantheon-systems/pantheon-advanced-page-cache#202]
After upgrading to 1.1.1, an internal server error gets thrown with the following debug message when executing a graphql query:

> ReflectionClass::__construct(): Argument #1 ($objectOrClass) must be of type object|string, array given

Likely caused by the `graphql_dataloader_get_model` filter passing a mixed data type, in this case `array(12)`, but the `Pantheon_Advanced_Page_Cache\Emitter::filter_graphql_dataloader_get_model` method not handling that before using it to construct a `ReflectionClass`.

Here's a partial stack trace:

```
array(3) {
  [“errors”]=&gt;
  array(1) {
    [0]=&gt;
    array(6) {
      [“debugMessage”]=&gt;
      string(103) “ReflectionClass::__construct(): Argument #1 ($objectOrClass) must be of type object|string, array given”
      [“message”]=&gt;
      string(21) “Internal server error”
      [“extensions”]=&gt;
      array(1) {
        [“category”]=&gt;
        string(8) “internal”
      }
      [“locations”]=&gt;
      array(1) {
        [0]=&gt;
        array(2) {
          [“line”]=&gt;
          int(4)
          [“column”]=&gt;
          int(9)
        }
      }
      [“path”]=&gt;
      array(2) {
        [0]=&gt;
        string(4) “cart”
        [1]=&gt;
        string(8) “contents”
      }
      [“trace”]=&gt;
      array(47) {
        [0]=&gt;
        array(3) {
          [“file”]=&gt;
          string(74) “/app/wp-content/plugins/pantheon-advanced-page-cache/inc/class-emitter.php”
          [“line”]=&gt;
          int(388)
          [“call”]=&gt;
          string(39) “ReflectionClass::__construct(array(12))”
        }
        [1]=&gt;
        array(3) {
          [“file”]=&gt;
          string(34) “/app/wp-includes/class-wp-hook.php”
          [“line”]=&gt;
          int(310)
          [“call”]=&gt;
          string(84) “Pantheon_Advanced_Page_Cache\Emitter::filter_graphql_dataloader_get_model(array(12))”
        }
        [2]=&gt;
        array(3) {
          [“file”]=&gt;
          string(27) “/app/wp-includes/plugin.php”
          [“line”]=&gt;
          int(205)
          [“call”]=&gt;
          string(43) “WP_Hook::apply_filters(array(12), array(4))”
        }
        [3]=&gt;
        array(3) {
          [“file”]=&gt;
          string(73) “/app/wp-content/plugins/wp-graphql/src/Data/Loader/AbstractDataLoader.php”
          [“line”]=&gt;
          int(411)
          [“function”]=&gt;
          string(163) “apply_filters(‘graphql_dataloader_get_model’, array(12), array(12), ‘f1298750ed09618717f9c10ea8d1d3b0’, instance of WPGraphQL\WooCommerce\Data\Loader\WC_Db_Loader)”
        }
        [4]=&gt;
        array(3) {
          [“file”]=&gt;
          string(73) “/app/wp-content/plugins/wp-graphql/src/Data/Loader/AbstractDataLoader.php”
          [“line”]=&gt;
          int(135)
          [“call”]=&gt;
          string(104) “WPGraphQL\Data\Loader\AbstractDataLoader::normalize_entry(array(12), ‘f1298750ed09618717f9c10ea8d1d3b0’)”
        }
        [5]=&gt;
        array(3) {
          [“file”]=&gt;
          string(85) “/app/wp-content/plugins/wp-graphql/src/Data/Connection/AbstractConnectionResolver.php”
          [“line”]=&gt;
          int(417)
          [“call”]=&gt;
          string(82) “WPGraphQL\Data\Loader\AbstractDataLoader::load(‘f1298750ed09618717f9c10ea8d1d3b0’)”
        }
        [6]=&gt;
        array(3) {
          [“file”]=&gt;
          string(85) “/app/wp-content/plugins/wp-graphql/src/Data/Connection/AbstractConnectionResolver.php”
          [“line”]=&gt;
          int(768)
          [“call”]=&gt;
          string(104) “WPGraphQL\Data\Connection\AbstractConnectionResolver::get_node_by_id(‘f1298750ed09618717f9c10ea8d1d3b0’)”
        }
        [7]=&gt;
        array(3) {
          [“file”]=&gt;
          string(85) “/app/wp-content/plugins/wp-graphql/src/Data/Connection/AbstractConnectionResolver.php”
          [“line”]=&gt;
          int(1002)
          [“call”]=&gt;
          string(65) “WPGraphQL\Data\Connection\AbstractConnectionResolver::get_nodes()”
        }
        [8]=&gt;
        array(3) {
          [“file”]=&gt;
          string(106) “/app/wp-content/plugins/wp-graphql/vendor/webonyx/graphql-php/src/Executor/Promise/Adapter/SyncPromise.php”
          [“line”]=&gt;
          int(67)
          [“call”]=&gt;
          string(91) “WPGraphQL\Data\Connection\AbstractConnectionResolver::WPGraphQL\Data\Connection\{closure}()”
        }
        [9]=&gt;
        array(3) {
          [“file”]=&gt;
          string(106) “/app/wp-content/plugins/wp-graphql/vendor/webonyx/graphql-php/src/Executor/Promise/Adapter/SyncPromise.php”
          [“line”]=&gt;
          int(53)
          [“call”]=&gt;
          string(90) “GraphQL\Executor\Promise\Adapter\SyncPromise::GraphQL\Executor\Promise\Adapter\{closure}()”
        }
        [10]=&gt;
        array(3) {
          [“file”]=&gt;
          string(113) “/app/wp-content/plugins/wp-graphql/vendor/webonyx/graphql-php/src/Executor/Promise/Adapter/SyncPromiseAdapter.php”
          [“line”]=&gt;
          int(149)
          [“call”]=&gt;
          string(56) “GraphQL\Executor\Promise\Adapter\SyncPromise::runQueue()”
        }
        [11]=&gt;
        array(3) {
          [“file”]=&gt;
          string(77) “/app/wp-content/plugins/wp-graphql/vendor/webonyx/graphql-php/src/GraphQL.php”
          [“line”]=&gt;
          int(105)
          [“call”]=&gt;
          string(103) “GraphQL\Executor\Promise\Adapter\SyncPromiseAdapter::wait(instance of GraphQL\Executor\Promise\Promise)”
        }
        [12]=&gt;
        array(3) {
          [“file”]=&gt;
          string(50) “/app/wp-content/plugins/wp-graphql/src/Request.php”
          [“line”]=&gt;
          int(656)
          [“call”]=&gt;
          string(217) “GraphQL\GraphQL::executeQuery(instance of WPGraphQL\WPSchema, ‘
    {
      cart {
        contents {
          itemCount
        }
      }
    }
‘, null, instance of WPGraphQL\AppContext, null, null, null, array(29))”
        }
        [13]=&gt;
        array(3) {
          [“file”]=&gt;
          string(55) “/app/wp-content/plugins/wp-graphql/access-functions.php”
          [“line”]=&gt;
          int(75)
          [“call”]=&gt;
          string(28) “WPGraphQL\Request::execute()”
        }
```

**Comment by macbookandrew** [comment]
I’m running into the same problem, though this error message in the wp-graphql response is really misleading:

```jsonc
{
  "errors": [
    {
      "debugMessage": "Class Array does not exist",
      "message": "Internal server error",
      "extensions": {
        "category": "internal"
      },
      "locations": [
        {
          "line": 10,
          "column": 7
        }
      ],
      "path": [
        "login",
        "customer",
        "downloadableItems"
      ]
    }
  ],
  // data, extensions, etc.
}
```

**Comment by jspellman814** [comment]
Thanks for this report and the PR. We are tracking this internally as BUGS-5812. I don't have an ETA on that at the moment, though today I'll re-roll #208 so it can run against our test suite.

**Comment by jspellman814** [comment]
Hodwy folks 👋 the fix from #212 has been included in the 1.2.1 release. Please upgrade and let us know if you are still experiencing issues. Thanks!

**Comment by jspellman814** [comment]
We're going to close this issue, please re-open if it persists. Thank you!
